rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java

```diff
@@ -1050,4 +1065,139 @@ private void isFreeBridgeTx(boolean expected, RskAddress destinationAddress, byt
     private Genesis getGenesisInstance(TrieStore trieStore) {
         return new TestGenesisLoader(trieStore, "frontier.json", constants.getInitialNonce(), false, true, true).load();
     }
+
+    private ErpFederation createErpFederation() {
+        BridgeConstants bridgeConstants = BridgeRegTestConstants.getInstance();
+        Federation genesisFederation = bridgeConstants.getGenesisFederation();
+        return new ErpFederation(
+            genesisFederation.getMembers(),
+            genesisFederation.getCreationTime(),
+            genesisFederation.getCreationBlockNumber(),
+            genesisFederation.getBtcParams(),
+            bridgeConstants.getErpFedPubKeysList(),
+            bridgeConstants.getErpFedActivationDelay()
+        );
+    }
+
+    private BtcTransaction createReleaseTx(
+        List<byte[]> signatures,
+        int inputsToAdd,
+        Federation federation,
+        boolean isFastBridge
+    ) {
+        // Setup
+        NetworkParameters btcParams = RegTestParams.get();
+        Address address;
+        byte[] program;
+
+        if (federation == null) {
+            federation = BridgeRegTestConstants.getInstance().getGenesisFederation();
+        }
+
+        if (isFastBridge) {
+            // Create fast bridge redeem script
+            Sha256Hash derivationArgumentsHash = Sha256Hash.of(new byte[]{1});
+
+            if (federation instanceof ErpFederation) {
+                Script erpFastBridgeRedeemScript =
+                    FastBridgeErpRedeemScriptParser.createFastBridgeErpRedeemScript(
+                        federation.getRedeemScript(),
+                        derivationArgumentsHash
+                    );
+
+                Script fastBridgeErpP2sh = ScriptBuilder.createP2SHOutputScript(erpFastBridgeRedeemScript);
+                address = Address.fromP2SHHash(btcParams, fastBridgeErpP2sh.getPubKeyHash());
+                program = erpFastBridgeRedeemScript.getProgram();
+            } else {
+                Script fastBridgeRedeemScript =
+                    FastBridgeRedeemScriptParser.createMultiSigFastBridgeRedeemScript(
+                        federation.getRedeemScript(),
+                        derivationArgumentsHash
+                    );
+
+                Script fastBridgeP2SH = ScriptBuilder
+                    .createP2SHOutputScript(fastBridgeRedeemScript);
+                address = Address.fromP2SHHash(btcParams, fastBridgeP2SH.getPubKeyHash());
+                program = fastBridgeRedeemScript.getProgram();
```

Comment:
This could be simplified as:
```suggestion
            Script fastBridgeRedeemScript;
            if (federation instanceof ErpFederation) {
                fastBridgeRedeemScript =
                    FastBridgeErpRedeemScriptParser.createFastBridgeErpRedeemScript(
                        federation.getRedeemScript(),
                        derivationArgumentsHash
                    );
           else {
                fastBridgeRedeemScript = 
                    FastBridgeRedeemScriptParser.createMultiSigFastBridgeRedeemScript(
                        federation.getRedeemScript(),
                        derivationArgumentsHash
                    );
            }
                
            Script fastBridgeP2SH = ScriptBuilder
                    .createP2SHOutputScript(fastBridgeRedeemScript);
            address = Address.fromP2SHHash(btcParams, fastBridgeP2SH.getPubKeyHash());
            program = fastBridgeRedeemScript.getProgram();
```

Reply:
Done